### PR TITLE
trimmed serendipity elements

### PR DESCRIPTION
**Pull request by rckirby**
We've added (and implemented over in Firedrake land) the trimmed serendipity spaces per
Gillette/Kloefkorn/Sanders (http://www.numdam.org/item/10.5802/smai-jcm.41.pdf).

This just updates the list of registered elements.

### CODE REVIEW

**Comment by jmv2009** [comment]
Not sure if it's going to work though in Fenics without dual basis.

**Comment by rckirby** [comment]
Fenics doesn’t support Bell or Argyris, either, but they still appear in the UFL registry.  The element list doesn’t require a UFL user to implement the entire set.

**Review by dham — APPROVED** [review]

File: ufl/finiteelement/elementlist.py [hunk 1]
```diff
@@ -200,6 +200,10 @@ def show_elements():
                   ("quadrilateral",))
 register_element2("BDMCF", 1, HDiv, "contravariant Piola", (1, None),
                   ("quadrilateral",))
+register_element2("SminusE", 1, HCurl, "covariant Piola", (1, None), cubes[1:3])
+register_element2("SminusF", 1, HDiv, "contravariant Piola", (1, None), cubes[1:2])
```
Comment (dham) [inline ufl/finiteelement/elementlist.py:204]: Is there a reason that `SminusF` is not allowed on Hexahedra?

**Review by dham — CHANGES_REQUESTED** [review]
See the issue with `SminusF` on Hex

File: ufl/finiteelement/elementlist.py (hunk 1 above)
Comment (rckirby) [reply ufl/finiteelement/elementlist.py:204]: After digging a bit, it looks like we have two attempts at this (from Cyrus + Justin) that have come in under different names and with slightly different implementations?  I think Cyrus put in SminusE/F and Justin put in SminusDiv/Curl.  The latter supports hexes in practice, but I don't see what the former shouldn't in theory (currently, it's disallowed in FIAT on hexes).

Comment (rckirby) [reply ufl/finiteelement/elementlist.py:204]: If these are the same math object, perhaps we should merge them/make one an alias and wipe out the FI(n)AT implementation of the less complete one?

**Comment by dham** [comment]
superceded by #195

**Pull request closed without merging**